### DISCUSSION TITLE
Retrieve relevant chunks

### DISCUSSION THREAD

**Discussion by AnhDucVu-Hust** [NVIDIA-NeMo/Guardrails#205] [Q&A]
In my config.py file, i have customed my provider to search with threshold to implement RAG. But when i see the prompt it still has additional data.

When I read the source code i think that this don't use my search imlemented. How to make action retrieve_relevant_chunk use my custom search provider?

**Comment by drazvan** [comment]
@AnhDucVu-Hust : can you share your `config.py`?
